### PR TITLE
Add degree symbol to charmap

### LINKED ISSUE #157: Add degree symbol to charmap
Add degree symbol to charmap, so we can use this library to show temperature.

```diff
diff --git a/luma/led_matrix/segment_mapper.py b/luma/led_matrix/segment_mapper.py
index 91c3447..d6a8e5f 100644
--- a/luma/led_matrix/segment_mapper.py
+++ b/luma/led_matrix/segment_mapper.py
@@ -6,6 +6,7 @@ _DIGITS = {
     ' ': 0x00,
     '-': 0x01,
     '_': 0x08,
+    '°': 0x63,
     '\'': 0x02,
     '0': 0x7e,
     '1': 0x30,
```

image

### PR DESCRIPTION
**Pull request by rm-hull**
Fixes #157

### CODE REVIEW

**Comment by rm-hull** [comment]
py27 build fails because of different unicode/utf-8 handling .. any ideas @thijstriemstra ?

**Comment by thijstriemstra** [comment]
oh the (co-)author should be added to contributors?

**Comment by thijstriemstra** [comment]
> py27 build fails because of different unicode/utf-8 handling 

let me check.

**Comment by thijstriemstra** [comment]
Took another look but can't figure it out, since I'm not sure what 'dot_muncher' does. Perhaps you can comment it line by line in this PR and it'll become more clear.

**Comment by rm-hull** [comment]
### Notes

Seems like python2 doesn't properly iterate over unicode characters (but python 3 does), and this is the root cause of the problem.

* https://stackoverflow.com/questions/46711888/how-to-properly-iterate-over-unicode-characters-in-python

```python
Python 2.7.15 (default, Jun 17 2018, 12:46:58) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = "29.12°C"
>>> len(x)
8
>>> list(x)
['2', '9', '.', '1', '2', '\xc2', '\xb0', 'C'] 
>>> y = u"29.12°C"
>>> len(y)
7
>>> list(y)
[u'2', u'9', u'.', u'1', u'2', u'\xb0', u'C']
```

vs.

```python3
Python 3.7.0 (default, Jun 29 2018, 20:13:13) 
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = "29.12°C"
>>> len(x)
7
>>> list(x)
['2', '9', '.', '1', '2', '°', 'C']
```

File: tests/test_segment_mapper.py [hunk 1]
```diff
@@ -62,3 +62,9 @@ def test_regular_empty_buf():
     buf = mutable_string("")
     results = regular(buf)
     assert list(results) == []
+
+
+def test_degrees():
+    buf = mutable_string(u"29.12°C")
```
[the remark this answers is not in the document]
Comment (thijstriemstra) [reply tests/test_segment_mapper.py]: alright. python 3 will work without I assume..

[the remark this answers is not in the document]
Comment (rm-hull) [reply tests/test_segment_mapper.py]: Just added a few more test scenarios ... who'd've thought from that simple patch that this would be so complicated 🤕

**Comment by thijstriemstra** [comment]
@rm-hull see this on py27 build now: https://travis-ci.org/rm-hull/luma.led_matrix/jobs/430275940#L775

```
=============================== warnings summary ===============================
/home/travis/build/rm-hull/luma.led_matrix/luma/led_matrix/segment_mapper.py:10: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  '\xb0': 0x63
/home/travis/build/rm-hull/luma.led_matrix/luma/led_matrix/segment_mapper.py:99: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  curr = _DIGITS.get(next(iterator), undefined)

**Comment by rm-hull** [comment]
hmmm - didn't notice that

**Pull request merged**